src/snowflake/snowpark/dataframe.py

```diff
@@ -3180,6 +3182,110 @@ def describe(self, *cols: Union[str, List[str]]) -> "DataFrame":
             subcalls=res_df._plan.api_calls.copy(),
         )
         return res_df
+    
+    def summary(self, *cols: Union[str, List[str]]) -> "DataFrame":
+        """
+        Computes basic statistics for numeric columns, which includes
+        ``count``, ``mean``, ``stddev``, ``min``, ``max``, ``25th percentile``, 
+        ``median`` and ``75th percentile``. If no columns are provided, this function 
+        computes statistics for all numerical or string columns. Non-numeric
+        and non-string columns will be ignored when calling this method.
+
+        Example::
+            >>> df = session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"])
+            >>> desc_result = df.summary().show()
+            -------------------------------------------------------
+            |"SUMMARY"  |"A"                 |"B"                 |
+            -------------------------------------------------------
+            |count      |2.0                 |2.0                 |
+            |max        |3.0                 |4.0                 |
+            |mean       |2.0                 |3.0                 |
+            |min        |1.0                 |2.0                 |
+            |stddev     |1.4142135623730951  |1.4142135623730951  |
+            |25%        |1.25                |3.25                |
+            |50%        |1.5                 |3.5                 |
+            |75%        |1.75                |3.75                |
+            -------------------------------------------------------
+            <BLANKLINE>
+
+        Args:
+            cols: The names of columns whose basic statistics are computed.
+        """
+        cols = parse_positional_args_to_list(*cols)
+        df = self.select(cols) if len(cols) > 0 else self
+
+        # ignore non-numeric and non-string columns
+        numerical_string_col_type_dict = {
+            field.name: field.datatype
+            for field in df.schema.fields
+            if isinstance(field.datatype, (StringType, _NumericType))
+        }
+
+        stat_func_dict = {
+            "count": count,
+            "mean": mean,
+            "stddev": stddev,
+            "min": min_,
+            "25%": approx_percentile,
```

Comment:
Let's use lambda here, i.e. "25%": lambda col: approx_percentile(col, 0.25) to simplify the code below.